cmd/zinject/zinject.c

```diff
@@ -208,6 +208,36 @@ type_to_name(uint64_t type)
 	}
 }
 
+struct errstr {
+	int		err;
+	const char	*str;
+};
+static const struct errstr errstrtable[] = {
+	{ EIO,		"io" },
+	{ ECKSUM,	"checksum" },
+	{ EINVAL,	"decompress" },
+	{ EACCES,	"decrypt" },
+	{ ENXIO,	"nxio" },
+	{ ECHILD,	"dtl" },
+	{ EILSEQ,	"corrupt" },
+};
+
+static int
+str_to_err(const char *str)
+{
+	for (int i = 0; i < sizeof (errstrtable); i++)
```

Comment:
I think this `sizeof (errstrtable)` might not be what you are looking for? I ran the following with this commit and I got a segfault:
`zinject -d nvme0n1 -e foo -T read -f 0.001 local_zpool `
The issue was the array bounds were exceed in `str_to_err()`.

Reply:
Oh yeah, sizeof was a brain fade, good catch.

I'll look into a test in a couple of days. Worth doing!